tests/test_runs/test_run_functions.py

```diff
@@ -754,6 +754,35 @@ def test_run_with_classifiers_in_param_grid(self):
         self.assertRaises(TypeError, openml.runs.run_model_on_task,
                           task=task, model=clf, avoid_duplicate_runs=False)
 
+    def test_run_with_illegal_flow_id(self):
+        # check the case where the user adds an illegal flow id to a non-existing flow
+        task = openml.tasks.get_task(1)
+        clf = DecisionTreeClassifier()
+        flow = sklearn_to_flow(clf)
+        flow, _ = self._add_sentinel_to_flow_name(flow, None)
+        flow.flow_id = -1
+
+        self.assertRaises(ValueError, openml.runs.run_flow_on_task,
```

Comment:
Could you please make this a `self.assertRaisesRegex`? The function is quite complicated, so it would be good to make sure it fails exactly where we expect it to fail.